### ISSUE TITLE
[VPAT][A11y] Datepicker disabled year and date is not readable by screen reader

### ISSUE THREAD

**Issue by chunhtai** [flutter/flutter#172988]
b/428648901
b/428825396

WCAG 1.3.2 suggest Meaningful content MUST be available to assistive technologies such as screen reader software

This is happen for both android and iOS

the disabled year button and date button is not focusable thus unreadable by screen reader

<img width="1011" height="1797" alt="Image" src="https://github.com/user-attachments/assets/7c06a114-50b5-43d2-a49b-9414b3e3badc" />

Expected 2023 year to be read as "2023, Button, disabled"

Same for date button

<img width="3839" height="1881" alt="Image" src="https://github.com/user-attachments/assets/43385c78-02ec-4c02-8e4a-bc7c59d2754b" />

**Comment by darshankawar** [comment]
Verified using given repro on Android and confirmed that the behavior persists with latest stable and master versions.

```
stable : 3.32.8
master : 3.33.0-1.0.pre-1200
```
